Docs/_extracli/BRAINSDWICleanup.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
For each remote module let's add a comment documenting which XML descriptions should be updated. For example, for `BRAINSTools`, such comment would be added here:

https://github.com/Slicer/Slicer/blob/7cce89a83f86ea9d542bae7b090912da422ce64b/SuperBuild.cmake#L314-L315

Including in the comment the set of bash command download the files would be ideal, something like this would do it for `BRAINSTools`. Similar comment would be added for DMRI, ... :

```bash
cd Slicer
for moduleName in \
  BRAINSDWICleanup \
  BRAINSDeface \
  BRAINSFit \
  BRAINSLabelStats \
  BRAINSROIAuto \
  BRAINSResample \
  BRAINSResize \
  BRAINSStripRotation \
  BRAINSTransformConvert;
do
  echo "Downloading ${moduleName}/${moduleName}.xml"
  curl -L# https://raw.githubusercontent.com/BRAINSia/BRAINSTools/590c371cecfc9e6f4dfed6dcc6416f67bccdaaf6/${moduleName}/${moduleName}.xml -o Docs/_extracli/${moduleName}.xml;
done
```

Reply:
Or (1) a script called `Utilities/Scripts/UpdateDocsExtraCLIDescription.sh` could be added  ... this script would extract the relevant `GIT_TAG`/`GIT_REPOSITORY` info from `SuperBuild.cmake`, and (2) add a comment to `SuperBuild.cmake` indicating this script should be executed.

That said this may be more error prone as it would rely on the exact syntax in the CMake file and be fragile ...

Reply:
We should have a solution that scales well for documenting all the extensions, not just those that are bundled with Slicer and that works for all module types, not just CLI. This would allow us to include documentation of _all_ extensions in the Slicer documentation, the same way as for bundled extensions. This would result in a documentation that is similar to the old wiki, where all the extensions are documented, not only Slicer core modules. Building the documentation should be a lightweight process and so it should not include installing or building Slicer - so that it can be easily handled by readthedocs.

This could be achieved for example by having all necessary information in the extension description files (in json). There could be a link to a markdown file for the extension and a markdown file for each module (or the markdown file content could be embedded in the json file) and for CLI modules there could be a link to an additional XML file.

Until we have a good design for this, I think we should go minimal - and do all updates manually. Since there is very small development activity in BRAINS and other bundled extensions, in particular in CLI modules, and especially at such level that would result in documentation changes, I would expect that we will not need to do any updates for 1-2 years (and by then we should have the more generic extension documentation infrastructure implemented).

Reply:
> I think we should go minimal - and do all updates manually

Agreed, in the mean time what about adding a comment like this near relevant remote extensions ? 

```
# Consider updating CLI description files found in Docs/_extracli directory
```

Reply:
Good idea, I'll add these comments.